pkg/apis/middleware/session.go

```diff
@@ -3,22 +3,24 @@ package middleware
 import (
 	"context"
 
-	"github.com/coreos/go-oidc"
 	sessionsapi "github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
 )
 
 // TokenToSessionFunc takes a rawIDToken and an idToken and converts it into a
 // SessionState.
-type TokenToSessionFunc func(ctx context.Context, rawIDToken string, idToken *oidc.IDToken) (*sessionsapi.SessionState, error)
+type TokenToSessionFunc func(ctx context.Context, token string, verify VerifyFunc) (*sessionsapi.SessionState, error)
```

Comment:
This opens up the doors to the chain-able Bearer Token enabled providers in structured configs.

Also allows a hacky `Basic` provider that can only implement `CreateSessionFromToken` and we can potentially move the Basic auth codepaths into this chain in the future.